src/scaladoc/scala/tools/nsc/doc/model/ModelFactoryImplicitSupport.scala

```diff
@@ -171,6 +171,20 @@ trait ModelFactoryImplicitSupport {
         return Nil
       }
 
+      if (!settings.docImplicitsShowAll && viewSimplifiedType.resultType.typeSymbol == sym) {
```

Comment:
Another pattern is adding covariance or contravariance for an invariant type parameter:
```
  case class C[T](t: T)
  implicit addCovariance[X, Y >: X](c: C[X]): C[Y] = c.asInstanceOf[C[Y]]
```
Do you also want to remove this case?